cns/restserver/nodesubnet.go

```diff
@@ -0,0 +1,59 @@
+package restserver
+
+import (
+	"context"
+	"net/netip"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	nodesubnet "github.com/Azure/azure-container-networking/cns/nodesubnet"
+	"github.com/Azure/azure-container-networking/cns/types"
+	errors "github.com/pkg/errors"
+)
+
+var _ nodesubnet.IPConsumer = &HTTPRestService{}
+
+// Implement the UpdateIPsForNodeSubnet method for HTTPRestService
+func (service *HTTPRestService) UpdateIPsForNodeSubnet(secondaryIPs []netip.Addr) error {
+	secondaryIPStrs := make([]string, len(secondaryIPs))
+	for i, ip := range secondaryIPs {
+		secondaryIPStrs[i] = ip.String()
+	}
+
+	networkContainerRequest := nodesubnet.CreateNodeSubnetNCRequest(secondaryIPStrs)
+
+	code, msg := service.saveNetworkContainerGoalState(*networkContainerRequest)
+	if code != types.Success {
+		logger.Debugf("Error in processing IP change")
+		return errors.Errorf("failed to save fetched ips. code: %d, message %s", code, msg)
```

Comment:
I am very much of the mindset that you should log or return errors--not both. Since we _can_ return an error here, I lean more towards just doing that. We can add more contextual information (as you've done), and it can be someone else's responsibility towards the top of the call stack to log all the errors (usually the HTTP handler).

Reply:
Done.